app/services/segment_delete_service.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class SegmentDeleteService
+  SEGMENT_URI = URI('https://gdpr.segment.com/graphql')
+
+  class SegmentDeleteServiceError < StandardError; end
+  class MissingAuthenticationData < SegmentDeleteServiceError; end
+  class SegmentAuthenticationError < SegmentDeleteServiceError; end
+  class SegmentUserDeletionError < SegmentDeleteServiceError; end
+
+  # TODO: password in ENV/config :) and raise error or show warning message if it is not there
+
+  def initialize(user_id)
+    @user_id = user_id
+    @errors = ActiveModel::Errors.new(self)
+  end
+
+  attr_reader :user_id, :errors
+
+  def delete_user!
```

Comment:
This is the only purpose of this service, right? Can we rename `delete_user!` and `delete_user` respectively to `call!` and `call`?

Reply:
sure 👍